panther_battery/src/adc_node.py

```diff
@@ -59,72 +63,75 @@ def __init__(self, name: str) -> None:
         rospy.loginfo(f'[{rospy.get_name()}] Node started')
 
     def _motor_controllers_state_cb(self, driver_state: DriverState) -> None:
-        self._V_driv_front = driver_state.front.voltage
-        self._V_driv_rear = driver_state.front.current
-        self._I_driv_front = driver_state.rear.voltage
-        self._I_driv_rear = driver_state.rear.current
+        with self._lock:
+            self._V_driv_front = driver_state.front.voltage
+            self._V_driv_rear = driver_state.front.current
+            self._I_driv_front = driver_state.rear.voltage
+            self._I_driv_rear = driver_state.rear.current
 
     def _io_state_cb(self, io_state: IOState) -> None:
-        self._charger_connected = io_state.charger_connected
+        with self._lock:
+            self._charger_connected = io_state.charger_connected
```

Comment:
The variable _charger_connected is read within the _publish_battery_msg method, so we can utilize a `_lock` there.

Reply:
Added